R/get_advanced_match_stats.R

```diff
@@ -1,3 +1,42 @@
+#' @importFrom xml2 xml_find_all xml_attr xml_text
+#' @importFrom dplyr mutate
+.add_player_href <- function(df, parent_element, player_xpath) {
+  player_elements <- xml2::xml_find_all(parent_element, player_xpath)
+  players <- setNames(
+    xml2::xml_attr(player_elements, "href"),
+    xml2::xml_text(player_elements)
+  )
+  res <- dplyr::mutate(
+    df,
+    "Player_Href" = players[df$Player],
+    .after = "Player"
+  )
+  return(res)
+}
```

Comment:
this is the improvement made by this PR. other changes are refactors, moving functions in functions to the top-level of the script file (for easier debugging)